src/main/java/com/minecolonies/coremod/colony/Structures.java

```diff
@@ -590,6 +595,94 @@ public static void setMD5s(final Map<String, String> md5s)
         }
     }
 
+    /**
+     * Handle a schematic which has been cut into pieces.
+     * This method is valid on the server
+     * The schematic will be gathered until all pieces have been put together and then handled like on the client.
+     *
+     * @param bytes representing the schematic.
+     * @param id UUID.
+     * @param piece the piece.
+     * @param pieces the amount of pieces.
+     */
+    public static boolean handleSaveSchematicMessage(final byte[] bytes, final UUID id, final int pieces, final int piece)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 7, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to reduce its Cognitive Complexity from 28 to the 20 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)